internal/configs/named_values.go

```diff
@@ -123,6 +125,19 @@ func decodeVariableBlock(block *hcl.Block, override bool) (*Variable, hcl.Diagno
 		v.SensitiveSet = true
 	}
 
+	if attr, exists := content.Attributes["deprecated"]; exists {
+		valDiags := gohcl.DecodeExpression(attr.Expr, nil, &v.Deprecated)
+		diags = append(diags, valDiags...)
+		if !valDiags.HasErrors() && strings.TrimSpace(v.Deprecated) == "" {
+			diags = append(diags, &hcl.Diagnostic{
+				Severity: hcl.DiagError,
+				Summary:  "Invalid deprecated value",
+				Detail:   fmt.Sprintf("The variable name %q 'deprecated' field is defined but is empty.", v.Name),
+				Subject:  &block.LabelRanges[0],
+			})
```

Comment:
We had a small discussion around the messaging in https://github.com/opentofu/opentofu/pull/2633#discussion_r2056387265

I would suggest to have similar messages in both cases (variables and outputs). This is from the deprecation of outputs:

```
Summary: "Invalid `deprecated` attribute"
Detail: The "deprecated" argument must not be empty, and should provide instructions on how to migrate away from usage of this deprecated output value.`
```

Reply:
Applied in 68df3ff06c44a207e60b6ccf6c31af26594b6cfe.
Thanks for the suggestion.